src/lib/components/Swap/RoutingDiagram/index.tsx

```diff
@@ -0,0 +1,160 @@
+import TokenImg from 'lib/components/TokenImg'
+import styled, { ThemedText } from 'lib/theme'
+import { useMemo } from 'react'
+import { InterfaceTrade } from 'state/routing/types'
+
+import { getTokenPath, RoutingDiagramEntry } from './utils'
+
+const Wrapper = styled(Column)`
+  padding: 0.25em;
+`
+
+const RouteRow = styled(Row)`
+  grid-template-columns: 1em 1.15em 1fr 1em;
+  min-width: 430px;
+`
+
+const RouteDetailsContainer = styled(Row)`
+  padding: 0.1rem 0.5rem;
+  position: relative;
+`
+
+const ShortDottedLine = styled.div`
+  align-items: center;
+  display: flex;
+  opacity: 0.5;
+  overflow: hidden;
+  width: 71px;
+  z-index: 1;
+`
+
+const DottedLine = styled.div`
+  align-items: center;
+  display: flex;
+  opacity: 0.5;
+  position: absolute;
+  width: calc(100% - 1em);
+  z-index: 1;
+`
+
+const DotColor = styled(DotLine)`
+  path {
+    stroke: ${({ theme }) => theme.secondary};
+  }
+`
+
+const BaseBadge = styled(Row)`
+  background-color: ${({ theme }) => theme.outline};
+  border-radius: 0.5em;
+  grid-gap: 0.375em;
+  padding: 0.25em 0.375em;
+  z-index: 2; // To cover the dotted line.
+`
+
+// Used to cutoff space between badges and dotted lines.
+const TransparentBadgeWrapper = styled(BaseBadge)`
+  background-color: ${({ theme }) => theme.dialog};
+  padding: 0 4px;
+`
+
+const VersionBadge = styled(BaseBadge)`
+  background-color: ${({ theme }) => theme.module};
+  border-radius: 0.25em;
+  padding: 0 0.125em;
+`
+
+const DetailsRow = styled(Row)`
+  display: grid;
+  grid-template-columns: 4.8125em 1fr;
+  width: 100%;
+`
+
+const StyledAutoRouterLabel = styled(ThemedText.Body1)`
+  color: #27ae60;
+  font-weight: 500;
+  line-height: 1rem;
+  @supports (-webkit-background-clip: text) and (-webkit-text-fill-color: transparent) {
+    background-image: linear-gradient(90deg, #2172e5 0%, #54e521 163.16%);
+    -webkit-background-clip: text;
+    -webkit-text-fill-color: transparent;
+  }
+`
+
+const StyledAutoRouterIcon = styled(AutoRouterIcon)`
+  height: 0.875em;
+  width: 0.875em;
+  :hover {
+    filter: brightness(1.3);
+  }
+`
+
+function Pool({ currency0, currency1, feeAmount }: { currency0: Currency; currency1: Currency; feeAmount: FeeAmount }) {
+  return (
+    <TransparentBadgeWrapper>
+      <BaseBadge>
+        <Row>
+          <TokenImg token={currency0} />
+          <TokenImg token={currency1} />
+        </Row>
+        <ThemedText.Subhead1 fontSize={14}>{feeAmount / 10000}%</ThemedText.Subhead1>
+      </BaseBadge>
+    </TransparentBadgeWrapper>
+  )
+}
+
+export default function RoutingDiagram({ trade }: { trade: InterfaceTrade<Currency, Currency, TradeType> }) {
+  const routes: RoutingDiagramEntry[] = useMemo(() => getTokenPath(trade), [trade])
+
+  return (
+    <Wrapper gap={0.75}>
+      <Row justify="space-between">
+        <Row gap={0.25}>
+          <StyledAutoRouterIcon />
+          <StyledAutoRouterLabel fontSize={14}>Auto Router</StyledAutoRouterLabel>
```

Comment:
If you need to change the font size you should be using a different ThemedText. If there isn't an appropriate one ask Phil to make one or update the design to use one.